### PR TITLE
fix OverridableReleaseSpec EIP158 and adjust TxPriorityContractTests

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Fix tests
- This fixes the issue in #7252

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public OverridableReleaseSpec(IReleaseSpec spec)
 
         public bool IsEip155Enabled => _spec.IsEip155Enabled;
 
-        public bool IsEip158Enabled => _spec.IsEip158Enabled;
+        public bool IsEip158Enabled => _spec.IsEip1559Enabled;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs:58]: why like that? it looks weird and like a bug. Could you add comment?

File: src/Nethermind/Nethermind.AuRa.Test/Contract/TestContractBlockchain.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ protected TestContractBlockchain()
             SealEngineType = Nethermind.Core.SealEngineType.AuRa;
         }
 
-        public static async Task<TTest> ForTest<TTest, TTestClass>(string? testSuffix = null) where TTest : TestContractBlockchain, new()
+        public static async Task<TTest> ForTest<TTest, TTestClass>(string testSuffix = null) where TTest : TestContractBlockchain, new()
```
Comment (alexb5dh) [inline src/Nethermind/Nethermind.AuRa.Test/Contract/TestContractBlockchain.cs:24]: What is the purpose of this change? Quite sure most compilers/IDEs will show it as a warning. If really needed, using `null!` will get rid of the warning.

File: src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs:58]: LOL I messed up something with that PR, let me reopen

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs]: I messed up master an PR by accident and it showed the other way around, now it is proper, sorry :)

File: src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ public OverridableReleaseSpec(IReleaseSpec spec)
 
         public bool IsEip155Enabled => _spec.IsEip155Enabled;
 
-        public bool IsEip158Enabled => _spec.IsEip1559Enabled;
+        public bool IsEip158Enabled => _spec.IsEip158Enabled;
```
Comment (alexb5dh) [inline src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs:58]: Is existing logic a bug or was it intended for something?

**Review by alexb5dh — APPROVED** [review]

File: src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs (hunk 3 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Specs.Test/OverridableReleaseSpec.cs:58]: bug, but it was only used in tests and didn't really affect much of them except that one :)

**Comment by LukaszRozmej** [comment]
This changes are now included in #7252 and lets merge that one
